server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/models/agents/agents.js

```diff
@@ -275,9 +285,11 @@ define(['mithril', 'lodash', 'string-plus',
       agentConfigState: data.agent_config_state,
       agentState:       data.agent_state,
       buildState:       data.build_state,
-      resources:        data.resources,
+      resources:        defaultToEmptyIfNoResource(data.resources, []),
       environments:     data.environments,
-      buildDetails:     Agents.Agent.BuildDetails.fromJSON(data.build_details)
+      buildDetails:     Agents.Agent.BuildDetails.fromJSON(data.build_details),
+      elasticAgentId:   s.defaultToIfBlank(data.elastic_agent_id, ''),
+      elasticPluginId:  s.defaultToIfBlank(data.elastic_plugin_id, '')
```

Comment:
Please keep these elastic agent attributes `undefined`, a blank string still has some meaning associated with it.

That way the `isElastic` method simply becomes `_.isNil(self.elasticAgentId())`.